runtime/jvmti/suspendhelper.cpp

```diff
@@ -45,38 +45,48 @@ suspendThread(J9VMThread *currentThread, jthread thread, BOOLEAN allowNull, BOOL
 		if (NULL != targetThread)
 #endif /* JAVA_SPEC_VERSION >= 19 */
 		{
-			if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
+			if (0 != J9OBJECT_U32_LOAD(currentThread, threadObject, vm->isSuspendedInternalOffset)) {
 				rc = JVMTI_ERROR_THREAD_SUSPENDED;
 			} else {
-				if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_STOPPED)) {
-					/* Do not suspend dead threads. Mirrors SUN behaviour. */
-					rc = JVMTI_ERROR_THREAD_NOT_ALIVE;
+				if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
+					J9OBJECT_U32_STORE(currentThread, threadObject, vm->isSuspendedInternalOffset, 1);
 				} else {
```

Comment:
`isSuspendedInternal` only exists in JDK19+. This code won't work in older JDK versions. My recommendation is to not modify existing logic to prevent breaking older JDK versions. All new logic can go below where the else block exists.

Reply:
made changes in the suspend thread part as recommended to prevent breaking older JDK versions and added all new parts in the JDK-19 else block.